src/components/LiquidityDepth/hooks.ts

```diff
@@ -0,0 +1,91 @@
+import { createContext, useEffect, useState } from 'react'
+import { Currency } from '@uniswap/sdk-core'
+import { FeeAmount } from '@uniswap/v3-sdk'
+import JSBI from 'jsbi'
+import { usePoolTickData } from 'hooks/usePoolTickData'
+import { TickProcessed } from 'constants/ticks'
+
+export interface ChartEntry {
+  index: number
+  isCurrent: boolean
+  activeLiquidity: number
+  price0: number
+}
+
+export const ChartContext = createContext<
+  Partial<{ zoom: number; canZoomIn: boolean; canZoomOut: boolean; zoomIn: () => void; zoomOut: () => void }>
+>({})
+
+export function useDensityChartData({
+  currencyA,
+  currencyB,
+  feeAmount,
+}: {
+  currencyA: Currency | undefined
+  currencyB: Currency | undefined
+  feeAmount: FeeAmount | undefined
+}) {
+  const [formattedData, setFormattedData] = useState<ChartEntry[] | undefined>()
+  const [activeChartEntry, setActiveChartEntry] = useState<ChartEntry | undefined>()
+  const [maxLiquidity, setMaxLiquidity] = useState<JSBI>(JSBI.BigInt(0))
+
+  const { loading, error, activeTick, tickData } = usePoolTickData(currencyA, currencyB, feeAmount)
+
+  // clear data when inputs are cleared
+  useEffect(() => {
+    if (!currencyA || !currencyB || !feeAmount) {
+      setFormattedData(undefined)
+    }
+  }, [currencyA, currencyB, feeAmount])
+
+  useEffect(() => {
```

Comment:
Could we just memoize all this instead of setting it in state and clearing it on changes